scripts/tokenomics-emulation/ddc-smart-contract.ts

```diff
@@ -0,0 +1,55 @@
+import { KeyringPair } from "@polkadot/keyring/types";
+import { ExtrinsicStatus } from "@polkadot/types/interfaces";
+import { EventRecord } from "@polkadot/types/interfaces";
+import Network from "./network";
+import { ContractPromise } from "@polkadot/api-contract";
+import cere02Abi from "./contract/cere02-metadata.json";
+
+class DdcSmartContract {
+  private ddcContract: ContractPromise;
+
+  constructor(private readonly config: any, private readonly network: Network) {
```

Comment:
Can you pass only API here?

Reply:
In future, we may need to fetch the balance of an account to send the amount to the smart contract. So we need the whole Network calss.

Reply:
YAGNI.
In the future we will extend.

Reply:
okay.